pkg/controllers/vdb/upgrade.go

```diff
@@ -500,3 +502,69 @@ func (i *UpgradeManager) fetchOldImage() (string, bool) {
 	}
 	return "", false
 }
+
+func (i *UpgradeManager) traceActorReconcile(actor controllers.ReconcileActor) {
+	i.Log.Info("starting actor for upgrade", "name", fmt.Sprintf("%T", actor))
+}
+
+// isSubclusterIdle will run a query to see the number of connections
+// that are active for a given subcluster.  It returns a requeue error if there
+// are active connections still.
+func (i *UpgradeManager) isSubclusterIdle(ctx context.Context, pfacts *PodFacts, scName string) (ctrl.Result, error) {
+	pf, ok := pfacts.findPodToRunVsql(true, scName)
+	if !ok {
+		i.Log.Info("No pod found to run vsql.  Skipping active connection check")
+		return ctrl.Result{}, nil
+	}
+
+	sql := fmt.Sprintf(
+		"select count(session_id) sessions"+
```

Comment:
Any plans to use an endpoint to get the sessions?

Reply:
No plans at the moment. There is an endpoint, but I'm expecting this usage to be temporary. It is waiting on the server logic for pause/redirect that Fen is working on.